indico/modules/events/surveys/models/surveys.py

```diff
@@ -285,3 +285,9 @@ def send_submission_notification(self, submission):
 @listens_for(Survey.sections, 'remove')
 def _wrong_collection_modified(target, value, *unused):
     raise Exception('This collection is view-only. Use `items` for write operations!')
+
+
+@listens_for(Survey.items, 'append')
+@listens_for(Survey.items, 'remove')
+def _items_modified(target, value, *unused):
+    db.session.expire(target, ['questions', 'sections'])
```

Comment:
from sqlalchemy.orm import object_session
    object_session(target).expire(target, ['questions', 'sections'])

Usually it shouldn't matter but if someone ever accesses the objects from another session it should be expired from that session